its/src/test/java/com/sonar/it/csharp/TestProjectTest.java

```diff
@@ -28,36 +28,50 @@
 
 import static com.sonar.it.csharp.Tests.ORCHESTRATOR;
 import static com.sonar.it.csharp.Tests.getComponent;
+import static com.sonar.it.csharp.Tests.getMeasureAsInt;
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class TestProjectTest {
   @ClassRule
   public static final TemporaryFolder temp = TestUtils.createTempFolder();
+  private static final String CSHARP_ONLY_TEST_PROJECT = "TestOnlyProject";
+  // the below is explicitly marked as test with <SonarQubeTestProject> MSBuild property
+  private static final String EXPLICITLY_MARKED_AS_TEST = "HtmlCSharpExplicitlyMarkedAsTest";
 
-  private static final String PROJECT = "TestOnlyProject";
   private static BuildResult buildResult;
 
   @BeforeClass
   public static void init() throws Exception {
     TestUtils.reset(ORCHESTRATOR);
-    buildResult = Tests.analyzeProject(temp, PROJECT, null);
+    buildResult = Tests.analyzeProject(temp, CSHARP_ONLY_TEST_PROJECT, null);
   }
 
   @Test
   public void projectIsAnalyzed() {
-    assertThat(getComponent(PROJECT).getName()).isEqualTo("TestOnlyProject");
+    assertThat(getComponent(CSHARP_ONLY_TEST_PROJECT).getName()).isEqualTo("TestOnlyProject");
 
     assertThat(getComponent("TestOnlyProject:UnitTest1.cs").getName()).isEqualTo("UnitTest1.cs");
   }
 
   @Test
-  public void logsContainInfoAndWarning() {
-    assertThat(buildResult.getLogs()).contains(
-      "This C# sensor will be skipped, because the current solution contains only TEST files and no MAIN files. " +
```

Comment:
Like the comment for the GUI warning, we should keep a warning in the console when our product thinks there's only test code in a solution.

Example text: "The scanner detected only TEST files in the current solution. Your SonarQube/SonarCloud project will be missing many MAIN-code related issues. Read more..."